client/utils/rest.go

```diff
@@ -68,3 +87,134 @@ func WriteGenerateStdTxResponse(w http.ResponseWriter, txBldr authtxb.TxBuilder,
 }
 
 func urlQueryHasArg(url *url.URL, arg string) bool { return url.Query().Get(arg) == "true" }
+
+//----------------------------------------
+// Building / Sending utilities
+
+type BaseReq struct {
+	Name          string `json:"name"`
+	Password      string `json:"password"`
+	ChainID       string `json:"chain_id"`
+	AccountNumber int64  `json:"account_number"`
+	Sequence      int64  `json:"sequence"`
+	Gas           int64  `json:"gas"`
+	GasAdjustment string `json:"gas_adjustment"`
+}
+
+/*
+ReadRESTReq is a simple convenience wrapper that reads the body and
+unmarshals to the req interface.
+
+  Usage:
+    type SomeReq struct {
+      BaseReq            `json:"base_req"`
+      CustomField string `json:"custom_field"`
+    }
+    req := new(SomeReq)
+    err := ReadRESTReq(w, r, cdc, req)
+*/
+func ReadRESTReq(w http.ResponseWriter, r *http.Request, cdc *wire.Codec, req interface{}) error {
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+		return err
+	}
+	err = cdc.UnmarshalJSON(body, req)
+	if err != nil {
+		WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+		return err
+	}
+	return nil
+}
+
+func (req BaseReq) BaseReqValidate(w http.ResponseWriter) bool {
+	if len(req.Name) == 0 {
```

Comment:
Let's make sure all the string parameters from `BaseReq` are trimmed (eg in this case `strings.TrimSpace(req.Name)`). Otherwise all this validations will pass with blank strings